src/ui/ui_render/lib/merge_variables.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const ELIGIBLE_FLAT_MERGE_KEYS = ['uiCapabilities'];
+
+export function mergeVariables(
+  allDefaults: Record<string, any>,
+  ...incomingDefaults: Array<Record<string, any>>
+) {
+  return incomingDefaults.reduce((acc, defaults) => {
```

Comment:
Would you mind skipping the reduce and avoiding mutation of the `allDefaults` argument? I think I would be more obvious that `allDefaults` was being mutated if it wasn't being renamed to `acc` By the reduce, and also I feel like when we're just mutating a variable in a loop we should keep it simple and use `for...of`

Reply:
Sure, I can do that. I went the mutation route to be consistent with the original implementation  (via `_.defaults`), but I don't mind switching it up